osu.Game/Screens/Edit/Compose/Components/ComposeBlueprintContainer.cs

```diff
@@ -342,7 +346,8 @@ public HitObjectCompositionTool CurrentTool
 
                 currentTool = value;
 
-                refreshTool();
+                // As per stable editor, when changing tools, we should forcefully commit any pending placement.
+                commitIfPlacementActive();
```

Comment:
This doesn't seem 100% equivalent to the previous call. Previously, `refreshTool()` was called, which is `removePlacement()` and then `ensurePlacementCreated()`. Now, `commitIfPlacementActive()` only calls the first one of the two. Was this intended?

I can't see it causing any issue, but it is worrying me a bit here.

Reply:
Actually, on a second look, `Update()` calls `ensurePlacementCreated()` every frame anyway. So this likely doesn't matter.